connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -50,6 +50,7 @@
     private static final Logger log = LoggerFactory.getLogger(WorkerSourceTask.class);
 
     private static final long SEND_FAILED_BACKOFF_MS = 100;
+    private static final long UNPAUSE_AWAIT_TIMEOUT_MS = 100;
```

Comment:
Yeah, that's fair. We can change awaitUnpause() to block indefinitely and then let stop() call notifyAll(). Then we don't need this value.